include/librealsense2/dds/dds-device-server.cpp

```diff
@@ -0,0 +1,162 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2022 Intel Corporation. All Rights Reserved.
+
+#include <iostream>
+
+#include "dds-device-server.h"
+#include "dds-participant.h"
+
+#include <fastdds/dds/domain/DomainParticipant.hpp>
+#include <fastdds/dds/publisher/Publisher.hpp>
+#include <fastdds/dds/publisher/DataWriter.hpp>
+#include <fastdds/dds/publisher/qos/PublisherQos.hpp>
+#include <librealsense2/dds/topics/dds-topics.h>
+
+using namespace eprosima::fastdds::dds;
+using namespace librealsense::dds;
+
+
+class dds_device_server::dds_stream_server
+{
+public:
+    dds_stream_server( eprosima::fastdds::dds::DomainParticipant * participant,
+                       eprosima::fastdds::dds::Publisher * publisher,
+                       const std::string & topic_root,
+                       const std::string & stream_name )
+        : _participant( participant )
+        , _publisher( publisher )
+        , _topic( nullptr )
+        , _topic_type_ptr( std::make_shared< eprosima::fastdds::dds::TypeSupport >(
```

Comment:
This is the same for all streams, right? Any reason not to just have one and use it instead of creating it each time?

Reply:
Actually, the more I look at it, the more I don't like it. Why not just create it on the stack before we need it and then just get rid of it? Why do we need it as a member?

Reply:
Changed to be created on the stack for every stream for now